### PR TITLE
refacto package build/ publish

### PR DESCRIPTION
**Pull request by fahedouch**
Signed-off-by: fahed dorgaa <fahed.dorgaa@gmail.com>

### Description of your changes

Fixes #  

- Use bundled provider packages
- Switch to `UP` cmd when building provider package
- Support publish to xpkg.upbound.io

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -78,6 +79,8 @@ jobs:
       # point out linter violations).
       - name: Lint
         uses: golangci/golangci-lint-action@v3
+        env:
+           RUNNING_IN_CI: true
```
Comment (hasheddan) [inline .github/workflows/ci.yml:83]: @fahedouch how does this env var modify lint behavior?

File: Makefile [hunk 2]
```diff
@@ -20,13 +25,35 @@ GOLANGCILINT_VERSION = 1.49.0
...
-IMAGES = $(PROJECT_NAME) $(PROJECT_NAME)-controller
--include build/makelib/image.mk
+
+IMAGES = provider-ansible
+-include build/makelib/imagelight.mk
+
+# ====================================================================================
+# Setup XPKG
+
+XPKG_REG_ORGS ?= xpkg.upbound.io/crossplane index.docker.io/crossplane
```
Comment (hasheddan) [inline Makefile:44]: This is a recent update :)
```suggestion
XPKG_REG_ORGS ?= xpkg.upbound.io/crossplane-contrib index.docker.io/crossplanecontrib
```

File: Makefile [hunk 3]
```diff
@@ -20,13 +25,35 @@ GOLANGCILINT_VERSION = 1.49.0
...
+# NOTE(hasheddan): skip promoting on xpkg.upbound.io as channel tags are
+# inferred.
+XPKG_REG_ORGS_NO_PROMOTE ?= xpkg.upbound.io/crossplane
```
Comment (hasheddan) [inline Makefile:47]:
```suggestion
XPKG_REG_ORGS_NO_PROMOTE ?= xpkg.upbound.io/crossplane-contrib
```

File: package/crossplane.yaml [hunk 4]
```diff
@@ -4,12 +4,16 @@ metadata:
   name: provider-ansible
   annotations:
     meta.crossplane.io/maintainer: Crossplane Maintainers <info@crossplane.io>
-    meta.crossplane.io/source: provider-ansible
+    meta.crossplane.io/source: github.com/crossplane/provider-ansible
```
Comment (hasheddan) [inline package/crossplane.yaml:7]:
```suggestion
    meta.crossplane.io/source: github.com/crossplane-contrib/provider-ansible
```

**Review by hasheddan — COMMENTED** [review]
@fahedouch this is looking great! A few updates requested, but then we can ensure secrets are present and get a released published :)

File: Makefile (hunk 2 above)
Comment (fahedouch) [reply Makefile:44]: Yes I saw your post on slack :)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fahedouch) [reply .github/workflows/ci.yml]: fixed

**Comment by fahedouch** [comment]
@hasheddan I added a new commit to fix module path https://github.com/crossplane-contrib/provider-ansible/pull/116/commits/d5f557d1c1ad23e26673701e11d634ad9eadce44

**Review by hasheddan — APPROVED** [review]
I have updated secrets here so we should be good to publish :)

**Pull request merged**